sdk/personalizer/ai-personalizer-rest/review/ai-personalizer.api.md

```diff
@@ -0,0 +1,936 @@
+}
+
+// @public
+export interface EvaluationsGet200Response extends HttpResponse {
+    // (undocumented)
+    body: EvaluationOutput;
+    // (undocumented)
+    status: "200";
+}
+
+// @public
+export interface EvaluationsGetdefaultResponse extends HttpResponse {
+    // (undocumented)
+    body: ErrorResponseOutput;
+    // (undocumented)
+    status: string;
+}
+
+// @public (undocumented)
+export type EvaluationsGetParameters = RequestParameters;
+
+// @public (undocumented)
+export interface EvaluationsList {
+    get(options?: EvaluationsListParameters): StreamableMethod<EvaluationsList200Response>;
+    post(options: EvaluationsCreateParameters): StreamableMethod<EvaluationsCreate201Response | EvaluationsCreatedefaultResponse>;
+}
+
+// @public
+export interface EvaluationsList200Response extends HttpResponse {
+    // (undocumented)
+    body: Array<EvaluationOutput>;
+    // (undocumented)
+    status: "200";
+}
+
+// @public (undocumented)
+export type EvaluationsListParameters = RequestParameters;
+
+// @public (undocumented)
+export interface EventsActivate {
+    post(options?: EventsActivateParameters): StreamableMethod<EventsActivate204Response | EventsActivatedefaultResponse>;
+}
+
+// @public
+export interface EventsActivate204Response extends HttpResponse {
+    // (undocumented)
+    body: Record<string, unknown>;
+    // (undocumented)
+    status: "204";
+}
+
+// @public
+export interface EventsActivatedefaultResponse extends HttpResponse {
+    // (undocumented)
+    body: ErrorResponseOutput;
+    // (undocumented)
+    status: string;
+}
+
+// @public (undocumented)
+export type EventsActivateParameters = RequestParameters;
+
+// @public (undocumented)
+export interface EventsReward {
+    post(options: EventsRewardParameters): StreamableMethod<EventsReward204Response | EventsRewarddefaultResponse>;
+}
+
+// @public
+export interface EventsReward204Response extends HttpResponse {
+    // (undocumented)
+    body: Record<string, unknown>;
+    // (undocumented)
+    status: "204";
+}
+
+// @public (undocumented)
+export interface EventsRewardBodyParam {
+    body: RewardRequest;
+}
+
+// @public
+export interface EventsRewarddefaultResponse extends HttpResponse {
+    // (undocumented)
+    body: ErrorResponseOutput;
+    // (undocumented)
+    status: string;
+}
+
+// @public (undocumented)
+export interface EventsRewardMediaTypesParam {
+    contentType?: "application/json";
+}
+
+// @public (undocumented)
+export type EventsRewardParameters = EventsRewardMediaTypesParam & EventsRewardBodyParam & RequestParameters;
+
+// @public (undocumented)
+export interface InternalErrorOutput {
+    code?: string;
+    innererror?: InternalErrorOutput;
+}
+
+// @public (undocumented)
+export function isUnexpected(response: ServiceConfigurationUpdate200Response | ServiceConfigurationUpdatedefaultResponse): response is ServiceConfigurationUpdatedefaultResponse;
```

Comment:
Do users need this method?

Reply:
This is generated by autorest. So leaving as is.

Reply:
@HarshaNalluru, this method is used for type guarding RLC responses. For example

```ts
const response = await client.path("/foo").get();

// Response can be Foo200Response or FooDefaultResponse. In order to handle correctly the error scenario and get intelisense

// The generated isUnexpected has a map of operation => success codes to determine if it is expected or not
if (isUnexpected(response)) {
   // Here the type of response has been narrowed down to FooDefaultResponse
   throw new Error(response.body.error)
}

// Once we reach this point the type of response is Foo200Response
// so we get intelisense
console.log(response.body.someFooProperty);

```